include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -0,0 +1,102 @@
+// ============================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ============================================================================
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \brief Provides seqan3::view::kmer_hash.
+ */
+
+#pragma once
+
+#include <range/v3/view/sliding.hpp>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/std/view/all.hpp>
+#include <seqan3/std/view/transform.hpp>
+
+namespace seqan3::detail
+{
+// ============================================================================
+//  kmer_hash_fn (adaptor definition)
+// ============================================================================
+
+//!\brief View adaptor definition for view::kmer_hash.
+class kmer_hash_fn : public pipable_adaptor_base<kmer_hash_fn>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = pipable_adaptor_base<kmer_hash_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
```

Comment:
move this below the other stuff so you only have a single private and a single public section